core/src/main/java/org/opensearch/remote/metadata/client/AbstractSdkClient.java

```diff
@@ -248,4 +248,14 @@ protected boolean isGlobalResource(GetDataObjectResponse response) {
             .map(y -> Objects.equals(y, globalTenantId))
             .orElse(false);
     }
+
+    /**
+     * Throw exception when the request containing cmk. It should be used in Opensearch client.
```

Comment:
Can we make this more clear?  It looks to me like we're defaulting to an exception but subclasses should override this method to permit it (and only the OpenSearch client does this?)